File: src/pathfinder_combat_simulator/core/saving_throws.py
from dataclasses import dataclass
# It seems AbilityScores is needed here. Let's add the import.
# If it causes circular dependency issues later, we might need to refactor further
# or pass AbilityScores instance as a parameter instead of type hinting it directly in the class.
from .ability_scores import AbilityScores

@dataclass
class SavingThrows:
    """Represents saving throw bonuses"""
    fortitude_base: int = 0
    reflex_base: int = 0
    will_base: int = 0

    def calculate_save(self, save_type: str, ability_scores: AbilityScores) -> int:
        """Calculate total saving throw bonus"""
        base = getattr(self, f"{save_type}_base")
        if save_type == "fortitude":
            return base + ability_scores.get_modifier("constitution")
        elif save_type == "reflex":
            return base + ability_scores.get_modifier("dexterity")
        elif save_type == "will":
            return base + ability_scores.get_modifier("wisdom")
        return base

File: src/core/savingThrows.ts
import { AbilityScores } from "./abilityScores.ts";

export type SaveType = "fortitude" | "reflex" | "will";

export class SavingThrows {
  fortitude_base: number;
  reflex_base: number;
  will_base: number;

  constructor(
    fortitude_base: number = 0,
    reflex_base: number = 0,
    will_base: number = 0,
  ) {
    this.fortitude_base = fortitude_base;
    this.reflex_base = reflex_base;
    this.will_base = will_base;
  }

  calculateSave(save_type: SaveType, ability_scores: AbilityScores): number {
    let base: number;
    let ability_modifier: number;

    switch (save_type) {
      case "fortitude":
        base = this.fortitude_base;
        ability_modifier = ability_scores.getModifier("constitution");
        break;
      case "reflex":
        base = this.reflex_base;
        ability_modifier = ability_scores.getModifier("dexterity");
        break;
      case "will":
        base = this.will_base;
        ability_modifier = ability_scores.getModifier("wisdom");
        break;
      default:
        // Should not happen with SaveType, but as a fallback:
        console.warn(`Unknown save type: ${save_type}`);
        return 0;
    }
    return base + ability_modifier;
  }
}
